erpnext/assets/doctype/asset/asset.py

```diff
@@ -459,50 +459,63 @@ def get_default_finance_book_idx(self):
 	
 	def validate_make_gl_entry(self):
 		purchase_document = self.get_purchase_document()
-		asset_bought_with_invoice = purchase_document == self.purchase_invoice
-		fixed_asset_account, cwip_account = self.get_asset_accounts()
+		if not purchase_document:
+			return False
+
+		asset_bought_with_invoice = (purchase_document == self.purchase_invoice)
+		fixed_asset_account = self.get_fixed_asset_account()
+		
 		cwip_enabled = is_cwip_accounting_enabled(self.asset_category)
-		# check if expense already has been booked in case of cwip was enabled after purchasing asset
-		expense_booked = False
-		cwip_booked = False
+		cwip_account = self.get_cwip_account(cwip_enabled=cwip_enabled)
 
+		query = """SELECT name FROM `tabGL Entry` WHERE voucher_no = %s and account = %s"""
```

Comment:
@vishdha cant we do get doc with Filters instead of this query ?

Reply:
@vishdha could you make the changes